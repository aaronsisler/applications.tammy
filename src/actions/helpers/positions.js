export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
