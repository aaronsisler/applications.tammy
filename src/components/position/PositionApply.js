import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { startLogin } from 'Actions/auth';

export class PositionApply extends React.Component {
    constructor(props) {
        super(props);
    }

    handleLogin = () => this.props.startLogin('apply');

    render() {
        return (
            <div className="position_apply">
                {this.props.isAuthenticated &&
                    <div className="position_apply_content__apply">
                        <Link className="nav_link" to="apply">Apply Now</Link>
                    </div>
                }
                {!this.props.isAuthenticated &&
                    <div className="position_apply_content__login">
                        <button
                            className="button"
                            onClick={this.handleLogin}
                        >
                            Login with Google
                        </button>
                    </div>
                }
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state) => ({
    isAuthenticated: !!state.user
});

/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
    startLogin: (redirectUrl) => dispatch(startLogin(redirectUrl))
});

export default connect(mapStateToProps, mapDispatchToProps)(PositionApply);

PositionApply.propTypes = {
    isAuthenticated: PropTypes.bool,
    startLogin: PropTypes.func.isRequired,
};

