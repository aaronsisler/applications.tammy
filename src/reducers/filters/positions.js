const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
