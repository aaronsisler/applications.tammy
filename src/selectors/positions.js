// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
