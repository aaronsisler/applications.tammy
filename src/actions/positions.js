import database from 'Firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        return dispatch(setPositions(positions));
    });
