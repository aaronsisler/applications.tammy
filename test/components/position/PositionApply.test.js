import React from 'react';
import { shallow } from 'enzyme';
import { PositionApply } from 'Position/PositionApply';
import { positionId } from '../../fixtures/positions';

describe('PositionApply', () => {
    const startLogin = jest.fn();
    let wrapper;

    const buildWrapper = (isAuthenticated = true) => {
        wrapper = shallow(
            <PositionApply
                isAuthenticated={isAuthenticated}
                positionId={positionId}
                startLogin={startLogin}
            />
        );
    };

    describe('when user is Authenticated', () => {
        beforeEach(() => {
            buildWrapper();
        })

        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when user is NOT Authenticated', () => {
        beforeEach(() => {
            buildWrapper(false);
        })

        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        describe('when Login button is clicked', () => {
            beforeEach(() => {
                wrapper.find('button').simulate('click');
            });

            it('should call startLogin with apply redirectUrl', () => {
                expect(startLogin).toHaveBeenLastCalledWith(`/apply/${positionId}`);
            });
        });
    });
});
