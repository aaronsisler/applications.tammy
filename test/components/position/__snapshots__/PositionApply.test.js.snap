// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionApply when user is Authenticated should render correctly 1`] = `
<div
  className="position_apply"
>
  <div
    className="position_apply_content__apply"
  >
    <Link
      className="nav_link"
      onClick={[MockFunction]}
      replace={false}
      to="apply"
    >
      Apply Now
    </Link>
  </div>
</div>
`;

exports[`PositionApply when user is NOT Authenticated should render correctly 1`] = `
<div
  className="position_apply"
>
  <div
    className="position_apply_content__login"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
